src/Orleans/Serialization/SerializationManager.cs

```diff
@@ -1228,45 +1228,22 @@ public static void SerializeInner(object obj, BinaryTokenStreamWriter stream, Ty
 
         private static void WriteEnum(object obj, BinaryTokenStreamWriter stream, Type type)
         {
-            var underlyingType = Enum.GetUnderlyingType(type);
-            var size = Marshal.SizeOf(underlyingType);
-            switch (size)
-            {
-                case 1:
-                    stream.Write(Convert.ToByte(obj));
-                    break;
-                case 2:
-                    stream.Write(Convert.ToInt16(obj));
-                    break;
-                case 4:
-                    stream.Write(Convert.ToInt32(obj));
-                    break;
-                case 8:
-                    stream.Write(Convert.ToInt64(obj));
-                    break;
-                default:
-                    throw new NotSupportedException($"Serialization of type {type.GetParseableName()} is not supported.");
-            }
+            var t = Enum.GetUnderlyingType(type).TypeHandle;
+            if (t.Equals(byteTypeHandle) || t.Equals(sbyteTypeHandle)) stream.Write(Convert.ToByte(obj));
+            else if (t.Equals(shortTypeHandle) || t.Equals(ushortTypeHandle)) stream.Write(Convert.ToInt16(obj));
+            else if (t.Equals(intTypeHandle) || t.Equals(uintTypeHandle)) stream.Write(Convert.ToInt32(obj));
+            else if (t.Equals(longTypeHandle) || t.Equals(ulongTypeHandle)) stream.Write(Convert.ToInt64(obj));
+            else throw new NotSupportedException($"Serialization of type {type.GetParseableName()} is not supported.");
```

Comment:
According to the docs [here](https://msdn.microsoft.com/en-us/library/sbbt4032.aspx)
> The approved types for an enum are byte, sbyte, short, ushort, int, uint, long, or ulong.

I tested with char, just to be extra sure:
![image](https://cloud.githubusercontent.com/assets/203839/19134147/fe637ad0-8ba7-11e6-96e8-db2f55e5528b.png)